pkg/validator/shoot_validator.go

```diff
@@ -87,6 +87,23 @@ func (v *Shoot) validateShoot(valContext *validationContext) field.ErrorList {
 	allErrors = append(allErrors, gcpvalidation.ValidateWorkers(valContext.shoot.Spec.Provider.Workers, workersPath)...)
 	allErrors = append(allErrors, gcpvalidation.ValidateControlPlaneConfig(valContext.controlPlaneConfig, allowedZones, workersZones(valContext.shoot.Spec.Provider.Workers), controlPlaneConfigPath)...)
 
+	// WorkerConfig
+	for i, worker := range valContext.shoot.Spec.Provider.Workers {
+		if worker.ProviderConfig != nil {
+			workerConfig, err := decodeWorkerConfig(v.decoder, worker.ProviderConfig, workersPath.Index(i).Child("providerConfig"))
+			if err != nil {
+				allErrors = append(allErrors, field.Invalid(workersPath, err, "invalid providerConfig"))
+			}
+
+			var volumeType *string
+			if worker.Volume != nil {
+				volumeType = worker.Volume.Type
```

Comment:
You are only validating the `.spec.provider.workers[].volume.type`, but there are also the types of the data volumes in `.spec.provider.workers[].dataVolumes[].type`.
I didn't get it yet fully - can I specify `SCRATCH` volume type and local SSD interface only for the root volume or also for the data volumes?

Reply:
Sorry, it is my fault.
The type of datavolume should be verified.
I have updated the logic.